301-azure-firewall-dns-proxy/azuredeploy.json

```diff
@@ -0,0 +1,2743 @@
+      "metadata": {
+        "description": "Specify whether deploy a custom DNS forwarder in the Hub Virtual Network. Default value is false."
+      }
+    },
+    "dnsAvailabilitySetName": {
+      "type": "string",
+      "defaultValue": "[concat('availabilityset', uniqueString(resourceGroup().id))]",
+      "metadata": {
+        "description": "Name of the Availability Set used by the DNS virtual machine."
+      }
+    },
+    "dnsVmName": {
+      "type": "string",
+      "defaultValue": "DnsVm",
+      "metadata": {
+        "description": "Specifies the name of the DNS virtual machine."
+      }
+    },
+    "devVmName": {
+      "type": "string",
+      "defaultValue": "DevVm",
+      "metadata": {
+        "description": "Specifies the name of the virtual machine in the Development Virtual Network."
+      }
+    },
+    "prodVmName": {
+      "type": "string",
+      "defaultValue": "ProdVm",
+      "metadata": {
+        "description": "Specifies the name of the virtual machine in the Production Virtual Network."
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "defaultValue": "Standard_DS3_v2",
+      "metadata": {
+        "description": "Specifies the size of the virtual machine."
+      }
+    },
+    "imagePublisher": {
+      "type": "string",
+      "defaultValue": "Canonical",
+      "metadata": {
+        "description": "Specifies the image publisher of the disk image used to create the virtual machine."
+      }
+    },
+    "imageOffer": {
+      "type": "string",
+      "defaultValue": "UbuntuServer",
+      "metadata": {
+        "description": "Specifies the offer of the platform image or marketplace image used to create the virtual machine."
+      }
+    },
+    "imageSku": {
+      "type": "string",
+      "defaultValue": "18.04-LTS",
+      "allowedValues": [
+        "12.04.5-LTS",
+        "14.04.5-LTS",
+        "16.04.0-LTS",
+        "18.04-LTS"
+      ],
+      "metadata": {
+        "description": "Specifies the Ubuntu version for the VM. This will pick a fully patched image of this given Ubuntu version."
+      }
+    },
+    "authenticationType": {
+      "type": "string",
+      "defaultValue": "password",
+      "allowedValues": [
+        "sshPublicKey",
+        "password"
+      ],
+      "metadata": {
+        "description": "Specifies the type of authentication when accessing the Virtual Machine. SSH key is recommended."
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "defaultValue": "azadmin",
+      "metadata": {
+        "description": "Specifies the name of the administrator account of the virtual machine."
+      }
+    },
+    "adminPasswordOrKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Specifies the SSH Key or password for the virtual machine. SSH key is recommended."
+      }
+    },
+    "diskStorageAccounType": {
+      "type": "string",
+      "defaultValue": "Premium_LRS",
+      "allowedValues": [
```

Comment:
remove values that are not allowed for VM disks - the following are alloed

Premium_LRS 
Premium_ZRS   
StandardSSD_LRS
StandardSSD_ZRS  
Standard_LRS

Reply:
Fixed